FFMpegCore/FFMPEG/FFProbe.cs

```diff
@@ -76,9 +76,20 @@ private VideoInfo ParseVideoInfoInternal(VideoInfo info, string probeOutput)
             double videoSize = 0d;
             double audioSize = 0d;
 
-            var duration = TimeSpan.FromSeconds(double.TryParse((video ?? audio).Duration, NumberStyles.Any, CultureInfo.InvariantCulture, out var output) ? output : 0);
-            info.Duration = duration.Subtract(TimeSpan.FromMilliseconds(duration.Milliseconds));
-
+            string sDuration = (video ?? audio).Duration;
+            TimeSpan duration;
+            if (sDuration != null)
+            {
+                duration = TimeSpan.FromSeconds(double.TryParse(sDuration, NumberStyles.Any, CultureInfo.InvariantCulture, out var output) ? output : 0);
+            }
+            else
+            {
+                sDuration = (video ?? audio).Tags.Duration;
+                TimeSpan.TryParse(sDuration.Remove(sDuration.LastIndexOf('.')), CultureInfo.InvariantCulture, out duration);
+            }
+            // Strip milliseconds and additional ticks
+            info.Duration = new TimeSpan(duration.Days, duration.Hours, duration.Minutes, duration.Seconds);
```

Comment:
Why not just `info.Duration = duration`?

Reply:
Is it necessary to strip anything from the timespan?

Reply:
I wanted to keep the same granularity as in the original code (seconds), but we could of course increase the resolution and use the 'whole' TimeSpan?

Reply:
I think using the 'whole' TimeSpan would be better. The rest is just a matter of representation then :)

Reply:
Ok, short update; i changed it to `info.Duration = duration` and some tests failed
Interesting enough there are strange double rounding issues:
In the Video_Duration test:
Does not work
`Assert.AreEqual(video.Duration.TotalSeconds - 5, outputVideo.Duration.TotalSeconds, Double.Epsilon);`
Does work
```
Assert.AreEqual(video.Duration.Hours, outputVideo.Duration.Hours);
Assert.AreEqual(video.Duration.Minutes, outputVideo.Duration.Minutes);
Assert.AreEqual(video.Duration.Seconds - 5, outputVideo.Duration.Seconds);
Assert.AreEqual(video.Duration.Milliseconds, outputVideo.Duration.Milliseconds);
```

Maybe there are some Ticks off - but i am not sure if we can expect an encoded video to be millisecond or even tick perfekt

Reply:
Should we only check up to seconds?

Reply:
Yeah, I think that would make good sense.
As you say, we cannot expect them to be tick perfect